configure.ac

```diff
@@ -202,6 +204,114 @@ AS_CASE([$with_gc],
     )
 AC_MSG_RESULT([$with_gc])
 
+dnl
+dnl User setting: native thread-local storage (off by default)
+dnl
+
+AC_ARG_WITH([native-tls],
+    [AS_HELP_STRING([--with-native-tls],
+        [use native thread-local storage implementation])],
+        [],
+        [with_native_tls=no])
+AC_MSG_CHECKING([whether to use native tls])
+dnl
+dnl Distinguish between cases where we support __thread declarations
+dnl and situations where we use a pthread_getspecific() implementation.
+dnl Right now, we only do the latter for 64-bit macOS.
```

Comment:
This is better than what was there before, but still lacks information about the reasons: *why* don't we use `__thread` on macOS, resp. conversely: why do we  prefer it elsewhere over `pthread_getspecific()`.

All this should be explained once and in total in some places; other places then can refer to it: ".. for details on the reasons for this, please look at file XYZ"

Reply:
Yes, that's actually what I'm shooting for. I'm looking at putting a full explanation into tls.h, and then refer to that.